### DISCUSSION TITLE
REPL splash text

### DISCUSSION THREAD

**Discussion by cboostwvanooijen** [micropython/micropython#14357] [Core Development] [answered]
I am working on a rather large MicroPython library. The preferred way to use it is to include it in a MicroPython build as frozen code. This works fine, but I'd like to add something to the REPL start-up text to indicate that my library is include. Where in the code should I add this?

**Comment by dlech** [comment]
You can override `MICROPY_BANNER_NAME_AND_VERSION` in a board config file or by passing extra args to the make command `make ... CFLAGS_EXTRA='-DMICROPY_BANNER_MACHINE="..."'` (should be something similar for cmake-based ports as well).

**Reply by cboostwvanooijen** [reply, marked as answer]
Thanks! I will do it in a different way, but your search terms led me to the relevant part of the sources!

**Reply by cboostwvanooijen** [reply]
For the record, what I now do is

    WORKDIR /work/micropython/shared/runtime
    RUN sed -i '/#if MICROPY_PY_BUILTINS_HELP/amp_hal_stdout_tx_str("with godafoss 0.1\\r\\n");' pyexec.c
